src/components/editor/Store/create.ts

```diff
@@ -105,18 +124,58 @@ const getInitialState = <T>(
             );
         },
 
-        setStatus: (newVal) => {
+        setStatus: (value, path) => {
             set(
                 produce((state) => {
-                    state.isSaving = newVal === EditorStatus.SAVING;
-                    state.isEditing = newVal === EditorStatus.EDITING;
-                    state.status = newVal;
+                    if (Object.hasOwn(state.statuses, path)) {
+                        state.statuses[path] = value;
+                    } else {
+                        state.statuses = { ...state.statuses, [path]: value };
+                    }
```

Comment:
Same question here as in elsewhere in the PR... not totally sure why this approach is being taken.

Reply:
The answer is assistive technologies, my friend. _Hence_ the need for me to run through each file before merging, but I am not going to hold up a PR review for minor edits.